### ISSUE TITLE
Will the training code be released

### ISSUE THREAD

**Issue by zhanghongyong123456** [VITA-Group/SSHarmonization#6]
expect

**Comment by yueruchen** [comment]
The work was done when I did my intern at Adobe. Since I've already left adobe, it might be hard to provide a reproducible training recipe.

**Comment by zhanghongyong123456** [comment]
> The work was done when I did my intern at Adobe. Since I've already left adobe, it might be hard to provide a reproducible training recipe. But I'm willing to provide instruction here about some training details.

Thank you for your reply, 
1. Can you explain the details of the training,
2. Do you have a good recommendation algorithm for image harmonization？ I find that this direction is rarely

**Comment by yueruchen** [comment]
I've included all training details in the paper and you should first try to read that and then ask anything else if that is still confused. You can refer to the works who cite this work "https://arxiv.org/abs/1703.00069"

**Comment by wenching33** [comment]
@yueruchen  Hi, 
When I test the code, under one special condition the result is not as expected. The special condition is that reference image is an almost all dark clouded sky, and the input image is a bright beach scene. The expected output is darken beach scene, however, what I really got is a "red" shaded beach scene.
Now I am trying to re-train to fix this problem. But before that, can you give me some comment about why this happened? or any advice on training data selection when training a new model without this problem?
Thanks in advance.

**Comment by yueruchen** [comment]
This is interesting! I do have met similar problem before and it seems that the network collapse for some unknown reason. But I tuned some hyperpaprameter and fix it later(at least from my testing set). YourThis results may suggest that there is still some unstability in this corner case. I guess that is because my training set is not diverse enough and does not cover some strange point, like your reference image. You can finetune this model on a larger/more-diverse training set and see if it can be fixed.

**Comment by wenching33** [comment]
@yueruchen 
Hi, Would you explain what does "We adopt a scale jittering range of [256,320] and then crop a 224x224 image in the training stage."   mean?
This sentence is in 4.1 Implementation in your paper.  I don't understand what "scale jittering range of [256,320]" means.  Thanks in advance.

**Comment by yueruchen** [comment]
First random resize the image, resizing range is [256, 320], then crop 224x224 images

**Comment by wenching33** [comment]
@yueruchen 
"resizing range is [256, 320]", Does this mean both the width and height in target size range from 256 to 320?
So aspect ratio is not preserved, right?  
And cropped 224x224 images are then resized to 256x256 to feed in networks, am I right? Because, the input tensor size in inference code is 256x256.

**Comment by wenching33** [comment]
@yueruchen 
Thanks for your kind reply. I have another problem.
When training the model, the Loss distance term decrease to  zero very fast. But it seems the reason for that is weights
of the StyleEncoder becomes zeros, which should not be the correct solution.
Have you ever met similar situation？ Although I tried different initialzation like kaiming initialization. It still happened.
